presto-main/src/main/java/com/facebook/presto/operator/HashAggregationOperator.java

```diff
@@ -360,20 +568,159 @@ protected Page computeNext()
                             BlockBuilder output = pageBuilder.getBlockBuilder(types.size() + i);
                             aggregator.evaluate(groupId, output);
                         }
-
-                        groupId++;
                     }
 
                     return pageBuilder.build();
                 }
             };
         }
+
+        private void rebuildAggregators()
```

Comment:
I don't like the `rebuildPartialAggregators` because it is named after implementation and not function/purpose.